### PR TITLE
Create Next App should ship a cached `yarn.lock` file to speed up installs

### LINKED ISSUE #14713: Create Next App should ship a cached `yarn.lock` file to speed up installs
Create Next App can ship a prepared `yarn.lock` file (`yarn.lock.cached`) for instant resolving, reducing the time to create a project in about half.

See the Create React App CLI for inspiration.

### PR DESCRIPTION
**Pull request by khattakdev**
closes #14713

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| buildDuration | 13.3s | 13.3s | ⚠️ +14ms |
| nodeModulesSize | 66 MB | 66 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.263 | 2.353 | ⚠️ +0.09 |
| / avg req/sec | 1104.58 | 1062.37 | ⚠️ -42.21 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.477 | 1.602 | ⚠️ +0.13 |
| /error-in-render avg req/sec | 1692.32 | 1560.3 | ⚠️ -132.02 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..769f.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.2 kB | 57.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.68 kB | 5.68 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.09 kB | 7.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| index.html gzip | 951 B | 951 B | ✓ |
| link.html gzip | 960 B | 960 B | ✓ |
| withRouter.html gzip | 943 B | 943 B | ✓ |
| Overall change | 2.85 kB | 2.85 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| buildDuration | 14.5s | 13.8s | -624ms |
| nodeModulesSize | 66 MB | 66 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..769f.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.2 kB | 57.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.68 kB | 5.68 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.09 kB | 7.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 876 kB | 876 kB | ✓ |
| link.js | 915 kB | 915 kB | ✓ |
| routerDirect.js | 907 kB | 907 kB | ✓ |
| withRouter.js | 907 kB | 907 kB | ✓ |
| Overall change | 4.49 MB | 4.49 MB | ✓ |

</details>

</details>
Commit: 45a06fd76d7103bf86c281cfdc591bfab7a0e714

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| buildDuration | 11.7s | 12.3s | ⚠️ +537ms |
| nodeModulesSize | 66 MB | 66 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.053 | 2.057 | 0 |
| / avg req/sec | 1217.57 | 1215.63 | ⚠️ -1.94 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.269 | 1.2 | -0.07 |
| /error-in-render avg req/sec | 1970.46 | 2083.41 | +112.95 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..769f.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.2 kB | 57.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.68 kB | 5.68 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.09 kB | 7.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| index.html gzip | 951 B | 951 B | ✓ |
| link.html gzip | 960 B | 960 B | ✓ |
| withRouter.html gzip | 943 B | 943 B | ✓ |
| Overall change | 2.85 kB | 2.85 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| buildDuration | 12.6s | 12.6s | -52ms |
| nodeModulesSize | 66 MB | 66 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..769f.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.2 kB | 57.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.68 kB | 5.68 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.09 kB | 7.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js arsalan/14713 | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 876 kB | 876 kB | ✓ |
| link.js | 915 kB | 915 kB | ✓ |
| routerDirect.js | 907 kB | 907 kB | ✓ |
| withRouter.js | 907 kB | 907 kB | ✓ |
| Overall change | 4.49 MB | 4.49 MB | ✓ |

</details>

</details>
Commit: db1f8d9cfa63e8130add1fb2d4de5d6414256794

File: packages/create-next-app/create-app.ts [hunk 1]
```diff
@@ -164,6 +164,14 @@ export async function createApp({
       JSON.stringify(packageJson, null, 2) + os.EOL
     )
 
+    // Copy Yarn.lock file
+    if (useYarn) {
```
Comment (NMinhNguyen) [inline packages/create-next-app/create-app.ts:168]: Bear in mind you'll probably want something like https://github.com/facebook/create-react-app/pull/6253 to ensure this doesn't break private npm registry setups.

**Comment by timneutkens** [comment]
Going to close this as it's been addressed in different ways.

**Pull request closed without merging**